wp-content/plugins/tapestry/controller/class.tapestry-controller.php

```diff
@@ -55,6 +52,73 @@ public function updateTapestry($tapestry) {
         return $tapestry;
     }
 
+    /**
+     * Retrieve a Tapestry post
+     * 
+     * @return Object Tapestry
+     */
+    public function getTapestry() {
+        if (!$this->postId) {
+            return $this->_throwsError('INVALID_POST_ID');
+        }
+        return $this->_getTapestryById($postId);
```

Comment:
`$postId` is not defined here.

Reply:
Thanks a lot for pointing it out. My bad on this, I didn't refactor carefully.